### PR TITLE
Capture links in dynamic content

### PR DESCRIPTION
**Pull request by reneaaron**
### Describe the changes you have made in this PR

Capture ALL `lightning:` links.
 
### Link this PR to an issue

#1307 

### Type of change (Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Checklist

- [ ] My code follows the style guidelines of this project and performed a self-review of my own code
- [ ] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/extension/inpage-script/index.js [hunk 1]
```diff
@@ -105,7 +108,7 @@ if (document) {
           }
         });
     });
-  });
+  }, { capture: true });
```
Comment (bumi) [inline src/extension/inpage-script/index.js:111]: what does capture exactly do? 
and do you know what other options are maybe useful? (e.g. `passive`?

File: src/extension/inpage-script/index.js [hunk 2]
```diff
@@ -10,7 +10,10 @@ if (document) {
 
   // Intercept any `lightning:` requests
   window.addEventListener("click", (ev) => {
-    const target = ev.target;
+    // Use composedPath() for detecting links inside a Shadow DOM
```
Comment (escapedcat) [inline src/extension/inpage-script/index.js:13]: 🤯  🎉

**Review by bumi — COMMENTED** [review]
this is mainly the composedPath, correct?

File: src/extension/inpage-script/index.js [hunk 3]
```diff
@@ -9,106 +9,114 @@ if (document) {
...
-    let href;
-    let paymentRequest;
-    let lnurl;
-    let link; // used to dispatch a succcess event
+  window.addEventListener(
+    "click",
+    (ev) => {
+      // Use composedPath() for detecting links inside a Shadow DOM
+      // https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath
+      const target = ev.composedPath()[0] || ev.target;
```
[the remark this answers is not in the document]
Comment (reneaaron) [reply src/extension/inpage-script/index.js:17]: According to the docs, yes: 

image

**Comment by reneaaron** [comment]
> this is mainly the composedPath, correct?

I currently can't think of a case where composedPath is null. I'll test and remove the old `ev.target` potentially.

File: src/extension/inpage-script/index.js (hunk 1 above)
Comment (reneaaron) [reply src/extension/inpage-script/index.js]: Added some explanations here:

https://github.com/getAlby/lightning-browser-extension/issues/1307#issuecomment-1232778613

Went through the other options but couldn't find something that makes sense for us.

**Pull request merged**